### ISSUE TITLE
Translate all non-English words to English

### ISSUE THREAD

**Issue by marko-radosavljevic** [MyLibreLab/MyLibreLab#17]
We need to remove all non-English words from the codebase.
For example:    `public void amRasterAusrichten()` to `public void alignToTheGrid()`

I'm sure there are some great tools we can use to achieve this effortlessly, we don't need perfect translation at the moment, approximation is completely fine.

**Comment by Pratik-ahirrao** [comment]
Can i work on this issue?

**Comment by marko-radosavljevic** [comment]
Sure! ^^

**Comment by Pratik-ahirrao** [comment]
Can you tell me at which files i have to do that?

**Comment by marko-radosavljevic** [comment]
All of them. ^^
I wouldn't waste human resources on this tho. I would prefer some script to do the work en masse, and human just to review and make small changes before the merge. There has to be some fitting translation script?

**Comment by MaSven** [comment]
Some idea would be to extract first all String to an i18n file and after that do some autotranslation to get the first ground started.
Here is a link to a discussion about auto extracting https://stackoverflow.com/questions/1011423/is-there-a-semi-automated-way-to-perform-string-extraction-for-i18n

**Comment by LordRibblesdale** [comment]
Hi! I don't know if someone has a pull request ready, meanwhile I was preparing my own.
What does "binFertig" mean? In `StatusGummiBandXBackIF` interface. I found translation like "I am ready" or "I'm done" but I think it's something like "getPrepared"
Thanks in advance!

**Comment by weisJ** [comment]
It means "I’m done" but I’m not quite sure what it’s purpose is really. From the usage it looks like a sort of callback but what exactly has been completed I’m not sure.

**Comment by tsaglam** [comment]
> I wouldn't waste human resources on this tho. I would prefer some script to do the work en masse, and human just to review and make small changes before the merge. There has to be some fitting translation script, or text editor extension..?
> 
> Edit: [many vs code hits.](https://i.imgur.com/Mn1iHpx.png) I'm not familiar with those tools, maybe some of them can work?
> In short, we are searching for tool to help us solve our translation problem, because I believe there is a lot of non-English in the codebase currently.

I am not sure how well that will work for things like variable names etc. It might be the case that 99% of all reviewed autotranslations might be manually changed anyways. For example, I saw the discussion in the Discord regarding `bolBeschrAnzeigen` which is just `boolShowDescription`, but the auto-translation would be confused by `bol` and probably would not realize that `Beschr` is just short for Beschreibung (description). However, humans can easily understand that from context.

**Comment by marko-radosavljevic** [comment]
Welcome to the project @tsaglam! ^^

Yeah, I agree, I don't have a solution, but I think it would still greatly speed up translating work if there is just the diff you can quickly edit.

Any tool that would speed up translating work would be great. Smart regex that will pull out all the German words for example? Maybe that's an option.. I'm sure there is much more suitable software for translators, so please suggest any tools that you know.

In any case, all tools we use on the codebase **have** to be non-destructive. Toleration for information loss is 0. For example if there is a tool that will auto-translate, translation should be annotated or commented above the original.

**Comment by weisJ** [comment]
Because we have to restructure the code base anyway The easiest solution in my opinion would be to translate as we go along. Every time something is rewritten, that’s the chance to do the translation as well. There is no need to rush translate all of the code base at once.

**Comment by marko-radosavljevic** [comment]
I mostly agree, the issue I have with it is that you just can't quickly check some module/class/method and get information you need. If everyone translates every piece of code they read, commits/git history would be messy. Also, if nobody translates, there is duplicated work for each dev that reads the same file..

**Comment by LordRibblesdale** [comment]
I agree with @marko-radosavljevic for git history complexity (for example I'm working on #30, that is VisualLogic/Basis.java, that is approx 5k lines), maybe there will be multiple things to check, but I mostly agree with @weisJ about translating meanwhile working.
But I don't know if this could create a pull request problem when someone shares code that has not been translated (and the master code is).

**Comment by marko-radosavljevic** [comment]
Another big thing I forgot to mention about @weisJ suggestion is that I assume most devs can't translate German code well enough, so they would have to ask German native devs for help every time they want to translate something..

**Comment by marko-radosavljevic** [comment]
@Pratik-ahirrao are you still working on this? If you are, what's the status?

**Comment by marko-radosavljevic** [comment]
We have a new workflow for translation work! :tada: 

We will all open pull requests with translated files into [files-for-translation](https://github.com/MyLibreLab/MyLibreLab/tree/files-for-translation) branch, and track progress in #79
